pkgs/applications/editors/jetbrains/builder/default.nix

```diff
@@ -0,0 +1,85 @@
+# Builder for JetBrains IDEs (`mkJetBrainsProduct`)
+
+{
+  lib,
+  stdenv,
+  callPackage,
+
+  jdk,
+  fontconfig,
+  libGL,
+  libX11,
+
+  vmopts ? null,
+  forceWayland ? false,
+}:
+let
+  baseBuilder = if stdenv.hostPlatform.isDarwin then ./darwin.nix else ./linux.nix;
+  mkJetBrainsProductCore = callPackage baseBuilder { inherit vmopts; };
+in
+# Makes a JetBrains IDE
+{
+  pname,
+  src,
+  version,
+  buildNumber,
+  wmClass,
+  product,
+  productShort ? product,
+  meta,
+
+  libdbm,
+  fsnotifier,
+
+  extraWrapperArgs ? [ ],
+  extraLdPath ? [ ],
+  buildInputs ? [ ],
+  passthru ? { },
+}:
+mkJetBrainsProductCore {
+  inherit
+    pname
+    extraLdPath
+    jdk
+    src
+    version
+    buildNumber
+    wmClass
+    product
+    productShort
+    libdbm
+    fsnotifier
+    ;
```

Comment:
Off-topic refactoring suggestion:

Instead of explicitly inheriting everything, use the `@args` and `removeAttrs args excludeDrvArgNames //` pattern. This will work well when migrating to `lib.extendMkDerivation`.

This will:
1. preserve attr position information (`inherit` counts as defining a new attribute, `//` counts as re-using the original attribute)
2. reframe the args from "explicit inclusion" to "explicit exclusion", which usually makes more sense; most things _should_ be passed through implicitly to mkDerivation. The `excludeDrvArgNames` list highlights any that are not.